DemoqaProject/pageObjects/Elements.cs

```diff
@@ -7,10 +7,323 @@ namespace DemoqaProject.PageObjects
+            //Select specific category and verify
+            arrowButton.Click();
+            if (!desktopButton.Selected)
+            {
+                desktopButton.Click();
+                Console.WriteLine("Category Desktop selected");
+            }
+            else
+            {
+                Console.WriteLine("Category Desktop not selected");
+            }
+        }
+
+        public void SelectGroupAndUncheckSub()
+        {
+            ReadOnlyCollection<IWebElement> checkbox = driver.FindElements(By.XPath("//button[@class='rct-collapse rct-collapse-btn']"));
+            checkbox[1].Click();
+            if (!commandsButton.Selected)
+            {
+                commandsButton.Click();
+                Console.WriteLine("Sub group deselected");
+            }
+        }
+
+        public void ExpendAndCollapseAll()
+        {
+            expendAll.Click();
+            collapseAll.Click();
+        }
+
+        public void NavigateToRadioButton()
+        {
+            radioButton.Click();
+        }
+
+        public void CheckYesRadio()
+        {
+            yesRadio.Click();
+        }
+        
+        public void CheckImpressiveRadio()
+        {
+            impressiveRadio.Click();
+        }
+
+        public void NavigateToWebTables(string firstname, string lastname, string email, string age, string salary, string department)
+        {
+            WebTablesButton.Click();
+            AddButton.Click();
+            firstNameInput.SendKeys(firstname);
+            lastNameInput.SendKeys(lastname);
+            emailInput.SendKeys(email);
+            ageInput.SendKeys(age);
+            salaryInput.SendKeys(salary);
+            departmentInput.SendKeys(department);
+            submitForm.Click();
+        }
+
+        public void EditRecord()
+        {
+            editRecord.Click();
+            firstNameInput.Clear();
+            firstNameInput.SendKeys("Tom");
+            submitForm.Click();
+        }
+
+        public bool CheckName(string name)
+        {
+            ReadOnlyCollection<IWebElement> checkbox = driver.FindElements(By.XPath("//div[@class='rt-td']"));
+            return checkbox.Any(checkFirstname => checkFirstname.Text.Equals(name));
+        }
+
+        public void DeleteRecord()
+        {
+            deleteRecord.Click();
+        }
+
+        public void ChangeRows()
+        {
+            JSExecuter(rowChangeto5);
+            rowChangeto5.Click();
+            valueRows.Click();
+        }
+
+        public void ClickNext()
+        {
+            nextButton.Click();
+        }
+        
+        public void ClickPrevious()
+        {
+            previousButton.Click();
+        }
+
+        public void CheckSorting()
+        {
+            ReadOnlyCollection<IWebElement> checkbox = driver.FindElements(By.XPath("//div[@class='rt-resizable-header-content']"));
+            foreach (IWebElement el in checkbox)
+            {
+                Thread.Sleep(2000);
```

Comment:
!!! Please avoid using Thread.Sleep()
Use custom explicit waits for this purpose if you wait for something.

Reply:
I just used for debug purposes.